### PR TITLE
Skip UDP OOB tests on unsupported architectures

### PR DESCRIPTION
**Pull request by tmthrgd**
See https://github.com/miekg/dns/issues/660 for rationale and discussion.

### CODE REVIEW

File: udp_test.go [hunk 1]
```diff
@@ -74,6 +75,14 @@ func TestSetUDPSocketOptions(t *testing.T) {
 }
 
 func TestParseDstFromOOB(t *testing.T) {
+	if runtime.GOARCH != "amd64" {
```
Comment (miekg) [inline udp_test.go]: //build line in the file?

Comment (miekg) [reply udp_test.go]: yeah, it needs a new file.... I'll just merge as-is

**Comment by fastest963** [comment]
@tmthrgd can we use the ipv4/ipv6 package to just create the oob data to then test those methods? Would that get around the OS-dependent problem?

**Comment by tmthrgd** [comment]
@fastest963 That was my plan, but the test data puts IPv4 addresses into v6 control messages, which x/net expressly doesn’t support.

**Comment by fastest963** [comment]
@tmthrgd seems useful to do the other test cases just to verify that there isn't any non-amd64 breakage and just separate the test case with ipv4 in ipv6 specifically only on amd64. @miekg what do you think?

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [miekg/dns] Skip UDP OOB tests ..." ]
>@tmthrgd seems useful to do the other test cases just to verify that there isn't any non-amd64 breakage and just separate the test case with ipv4 in ipv6 specifically only on amd64. @miekg what do you think?

Are we testing it wrongly (from the viewpoint of x/net) ? If so, we should just 
update the test.

**Comment by fastest963** [comment]
Well x/net doesn't let you create a control message with a ipv4 address inside of a ipv6 control message, but Linux kernel definitely does that if you enable v4+v6 on an interface, it will send a ipv6 control message no matter if the source IP was ipv4 or not.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [miekg/dns] Skip UDP OOB tests ..." ]
>@tmthrgd seems useful to do the other test cases just to verify that there isn't any non-amd64 breakage and just separate the test case with ipv4 in ipv6 specifically only on amd64. @miekg what do you think?

sgtm, although I have no strong pref. for a new file with build or the check we 
have now.

**Pull request merged**